In [1]:
from langchain_text_splitters.character import CharacterTextSplitter

In [2]:
with open("some_data\\FDR_State_of_Union_1944.txt") as f:
    data = f.read()
len(data)

21995

In [24]:
len(data.split())

3750

In [25]:
splitter = CharacterTextSplitter(separator="\n\n", chunk_size = 1000)

In [26]:
documents = splitter.create_documents([data])

In [27]:
len(documents)

28

In [29]:
len(documents[0].page_content)   #843 characters

843

In [30]:
documents[0].page_content

"This Nation in the past two years has become an active partner in the world's greatest war against human slavery.\n\nWe have joined with like-minded people in order to defend ourselves in a world that has been gravely threatened with gangster rule.\n\nBut I do not think that any of us Americans can be content with mere survival. Sacrifices that we and our allies are making impose upon us all a sacred obligation to see to it that out of this war we and our children will gain something better than mere survival.\n\nWe are united in determination that this war shall not be followed by another interim which leads to new disaster- that we shall not repeat the tragic errors of ostrich isolationism—that we shall not repeat the excesses of the wild twenties when this Nation went for a joy ride on a roller coaster which ended in a tragic crash."

### This is text splitting, but text and token will not be same. Most of the llm deals with token size, so its better to split the text based on number tokens, the following shows the same

In [16]:
!pip install tiktoken


[notice] A new release of pip is available: 24.2 -> 25.0.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [18]:
splitter = CharacterTextSplitter.from_tiktoken_encoder(chunk_size = 500)
docs = splitter.split_text(data)

In [19]:
len(docs)

15

In [23]:
docs[0]

'This Nation in the past two years has become an active partner in the world\'s greatest war against human slavery.\n\nWe have joined with like-minded people in order to defend ourselves in a world that has been gravely threatened with gangster rule.\n\nBut I do not think that any of us Americans can be content with mere survival. Sacrifices that we and our allies are making impose upon us all a sacred obligation to see to it that out of this war we and our children will gain something better than mere survival.\n\nWe are united in determination that this war shall not be followed by another interim which leads to new disaster- that we shall not repeat the tragic errors of ostrich isolationism—that we shall not repeat the excesses of the wild twenties when this Nation went for a joy ride on a roller coaster which ended in a tragic crash.\n\nWhen Mr. Hull went to Moscow in October, and when I went to Cairo and Teheran in November, we knew that we were in agreement with our allies in o

In [22]:
len(docs[0].split())

411

### so you can see that 411 words creates almost 500 tokens, this is the benefit of using the token based split.